packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -181,6 +181,21 @@ export class GCFBootstrapper {
     return TriggerType.UNKNOWN;
   }
 
+  /**
+   * Binds the trigger info to the logger so that every logged statement
+   * includes the trigger info. Does not bind the message property.
+   *
+   * NOTE: statements logged before this method is called will not include
+   * the trigger info
+   * NOTE: this method modifies the triggerInfo parameter given
+   *
+   * @param triggerInfo trigger information for current execution
+   */
+  private static bindTriggerInfoToLogger(triggerInfo: TriggerInfo) {
+    delete triggerInfo.message;
```

Comment:
because `triggerInfo` passes by reference, this  will actually  remove the `message` parameter from the originating object,  this might not be the intended  behavior.

You could:

```js
const meta = {...triggerInfo}
logger = logger.child(meta);
```
```

Reply:
The removal of the property directly from the `triggerInfo` was intentional (I made a note of this in the method docs) because I thought cloning the object would be an expensive operation and right now we aren't using the `triggerInfo` object anywhere else after this method. 

However, if you think it might lead to some 'gotcha' situations in the future, I can use your suggestion above to clone it (or sometimes I even do `const meta = JSON.parse(JSON.stringify(triggerInfo))` for quick cloning)

Reply:
Also added an example to the PR description above -- will also add it to the issue

Reply:
I agree with @bcoe here on this one. Modifying the input object can lead to error prone code in the future. Also, the description of the function and name don't describe the modification of the input.

For performance optimizations like this, we should also consider the amount of memory saved in practice - the trigger info is really small (< 1KB) so this gain is probably negligible.

Reply:
If you do want to still delete the field in this manner, you can do it outside of this function in the caller.